### ISSUE TITLE
crash with failure to load gfx firmware and no console output on monitor

### ISSUE THREAD

**Issue by goproslowyo** [siderolabs/talos#8072]
### Description
I get no graphical output after the initrd and kernel begin the boot process. I can reach the nodes with talosctl and can use that to view the boot logs and found a crash during the boot process for the amd graphics firmware.

### Logs

The machines in question are Lenovo ThinkCentre M715q. They have AMD Radeon Graphics in processor via an AMD PRO A10-8770E R7, codename `carrizo`.

[https://support.lenovo.com/us/en/solutions/pd104468-detailed-specifications-and-overview-for-thinkcentre-m715q-tiny-form-factor](https://support.lenovo.com/us/en/solutions/pd104468-detailed-specifications-and-overview-for-thinkcentre-m715q-tiny-form-factor)
[https://download.lenovo.com/parts/ThinkCentre/m715q_tiny_spec.pdf](https://download.lenovo.com/parts/ThinkCentre/m715q_tiny_spec.pdf)
[https://www.techpowerup.com/cpu-specs/pro-a10-8770e.c2608](https://www.techpowerup.com/cpu-specs/pro-a10-8770e.c2608)

```
$ talosctl -n 192.168.2.205 dmesg
[...snip...]
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461233972Z]: amdgpu 0000:00:01.0: Direct firmware load for amdgpu/carrizo_pfp.bin failed with error -2
 SUBSYSTEM=pci
 DEVICE=+pci:0000:00:01.0
192.168.2.205: kern:     err: [2023-12-14T11:16:10.461245972Z]: amdgpu 0000:00:01.0: amdgpu: gfx8: Failed to load firmware "amdgpu/carrizo_pfp.bin"
 SUBSYSTEM=pci
 DEVICE=+pci:0000:00:01.0
192.168.2.205: kern:     err: [2023-12-14T11:16:10.461252972Z]: [drm:gfx_v8_0_sw_init.cold] *ERROR* Failed to load gfx firmware!
192.168.2.205: kern:     err: [2023-12-14T11:16:10.461265972Z]: [drm:amdgpu_device_init.cold] *ERROR* sw_init of IP block <gfx_v8_0> failed -2
192.168.2.205: kern:     err: [2023-12-14T11:16:10.461274972Z]: amdgpu 0000:00:01.0: amdgpu: amdgpu_device_ip_init failed
 SUBSYSTEM=pci
 DEVICE=+pci:0000:00:01.0
192.168.2.205: kern:     err: [2023-12-14T11:16:10.461280972Z]: amdgpu 0000:00:01.0: amdgpu: Fatal error during GPU init
 SUBSYSTEM=pci
 DEVICE=+pci:0000:00:01.0
192.168.2.205: kern:    info: [2023-12-14T11:16:10.461287972Z]: amdgpu 0000:00:01.0: amdgpu: amdgpu: finishing device.
 SUBSYSTEM=pci
 DEVICE=+pci:0000:00:01.0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461383972Z]: ------------[ cut here ]------------
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461387972Z]: WARNING: CPU: 0 PID: 1 at drivers/gpu/drm/amd/amdgpu/amdgpu_irq.c:655 amdgpu_irq_put+0x45/0x70
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461400972Z]: Modules linked in:
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461406972Z]: CPU: 0 PID: 1 Comm: swapper/0 Not tainted 6.1.61-talos #1
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461413972Z]: Hardware name: LENOVO 10M2S02Q00/30FD, BIOS M11KT56A 12/16/2022
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461418972Z]: RIP: 0010:amdgpu_irq_put+0x45/0x70
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461426972Z]: Code: 48 8b 4e 10 48 83 39 00 74 2c 89 d1 48 8d 04 88 8b 08 85 c9 74 14 f0 ff 08 b8 00 00 00 00 74 05 e9 f0 3d 16 01 e9 8b fd ff ff <0f> 0b b8 ea ff ff ff e9 df 3d 16 01 b8 ea ff ff ff e9 d5 3d 16 01
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461436972Z]: RSP: 0018:ffffa0a1c0037c28 EFLAGS: 00010246
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461442972Z]: RAX: ffff94ed83ce7620 RBX: ffff94ed84120000 RCX: 0000000000000000
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461448972Z]: RDX: 0000000000000000 RSI: ffff94ed84120be0 RDI: ffff94ed84120000
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461453972Z]: RBP: ffff94ed84120010 R08: 0000000000000000 R09: ffffa0a1c00379e0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461458972Z]: R10: 0000000000000003 R11: ffffffffae562d48 R12: ffff94ed84120010
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461463972Z]: R13: 0000000000000001 R14: ffff94ed84120010 R15: ffff94ed84120000
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461468972Z]: FS:  0000000000000000(0000) GS:ffff94f46f000000(0000) knlGS:0000000000000000
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461476972Z]: CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461481972Z]: CR2: ffff94f48efff000 CR3: 00000000a9010000 CR4: 00000000001506f0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461487972Z]: Call Trace:
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461492972Z]:  <TASK>
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461499972Z]:  ? __warn+0x7d/0xc0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461507972Z]:  ? amdgpu_irq_put+0x45/0x70
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461514972Z]:  ? report_bug+0xe6/0x170
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461521972Z]:  ? handle_bug+0x41/0x70
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461528972Z]:  ? exc_invalid_op+0x13/0x60
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461533972Z]:  ? asm_exc_invalid_op+0x16/0x20
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461541972Z]:  ? amdgpu_irq_put+0x45/0x70
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461548972Z]:  gmc_v8_0_hw_fini+0x17/0xa0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461555972Z]:  amdgpu_device_fini_hw+0x1cc/0x2af
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461561972Z]:  amdgpu_driver_load_kms.cold+0x54/0x6a
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461567972Z]:  amdgpu_pci_probe+0x125/0x340
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461574972Z]:  local_pci_probe+0x41/0x80
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461582972Z]:  pci_device_probe+0xbf/0x230
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461589972Z]:  ? kernfs_create_link+0x5d/0xa0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461597972Z]:  ? sysfs_do_create_link_sd+0x6e/0xe0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461604972Z]:  really_probe+0xc7/0x280
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461611972Z]:  ? pm_runtime_barrier+0x50/0x90
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461618972Z]:  __driver_probe_device+0x73/0xf0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461624972Z]:  driver_probe_device+0x1f/0x90
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461629972Z]:  __driver_attach+0x84/0x130
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461635972Z]:  ? __device_attach_driver+0xc0/0xc0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461641972Z]:  bus_for_each_dev+0x87/0xd0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461647972Z]:  bus_add_driver+0x186/0x1d0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461653972Z]:  driver_register+0x89/0xe0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461658972Z]:  ? drm_sched_fence_slab_init+0x87/0x87
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461666972Z]:  do_one_initcall+0x59/0x230
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461674972Z]:  kernel_init_freeable+0x2bb/0x35d
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461682972Z]:  ? rest_init+0xd0/0xd0
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461689972Z]:  kernel_init+0x16/0x130
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461695972Z]:  ret_from_fork+0x22/0x30
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461702972Z]:  </TASK>
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461705972Z]: ---[ end trace 0000000000000000 ]---
192.168.2.205: kern: warning: [2023-12-14T11:16:10.461984972Z]: amdgpu: probe of 0000:00:01.0 failed with error -2
```

### Environment

- Talos version: [`talosctl version --nodes <problematic nodes>`]

```shell
$ talosctl -n $(echo 192.168.2.20{1..5}|tr ' ' ,) version                                     
Client:
	Tag:         v1.5.5
	SHA:         ad7361c7
	Built:       
	Go version:  go1.20.11
	OS/Arch:     linux/amd64
Server:
	NODE:        192.168.2.203
	Tag:         v1.5.5
	SHA:         ad7361c7
	Built:       
	Go version:  go1.20.11
	OS/Arch:     linux/amd64
	Enabled:     RBAC
	NODE:        192.168.2.202
	Tag:         v1.5.5
	SHA:         ad7361c7
	Built:       
	Go version:  go1.20.11
	OS/Arch:     linux/amd64
	Enabled:     RBAC
	NODE:        192.168.2.204
	Tag:         v1.5.5
	SHA:         ad7361c7
	Built:       
	Go version:  go1.20.11
	OS/Arch:     linux/amd64
	Enabled:     RBAC
	NODE:        192.168.2.201
	Tag:         v1.5.5
	SHA:         ad7361c7
	Built:       
	Go version:  go1.20.11
	OS/Arch:     linux/amd64
	Enabled:     RBAC
	NODE:        192.168.2.205
	Tag:         v1.5.5
	SHA:         ad7361c7
	Built:       
	Go version:  go1.20.11
	OS/Arch:     linux/amd64
	Enabled:     RBAC
```

- Kubernetes version: [`kubectl version --short`]

```shell
$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.27.2
Kustomize Version: v5.0.1
Server Version: v1.28.3
```

- Platform:
Bare Metal

**Comment by smira** [comment]
you most probably need [amdgpu-firmware](https://github.com/siderolabs/extensions/tree/main/firmware/amdgpu-firmware) system extension

see [Boot Assets](https://www.talos.dev/v1.6/talos-guides/install/boot-assets/)

**Comment by goproslowyo** [comment]
Ah, yes, thank you @smira!

I was looking into that after I posted the issue last night. I actually went down this path but wasn't sure what to do once I ran into this error message: `error enhancing profile from schematic: official extension "siderolabs/amdgpu-firmware" is not available for Talos version v1.5.5`.

Screenshot:
![image](https://github.com/siderolabs/talos/assets/68455785/d5323e27-2668-46e6-8199-d1f9639e8666)
